### PR TITLE
HTML5 based login page

### PR DESCRIPTION
**Pull request by robertdown**
Drop the tables from the login page, replace them with bootstrap-compatible `div`s. Also updating the login logo to an SVG of the final color scheme.

Also added better default mini logos.

The page will look different based on if the user decides to display the extra logo. I'm attachign some screenshots to show the various possibilities
openemr_1_minilogo_only
openemr_all_logos
openemr_mini_extra
openemr_mini_logos
openemr_no_logos
openemr_xtra_only

Online demo is here:
http://www.open-emr.org/wiki/index.php/Development_Demo#192.168.1.132

### CODE REVIEW

**Comment by robertdown** [comment]
Also, I basically neutralized the login screen and added a `login` class to the `body` tag as well as cleaning out most of the login.css file so each theme can control the look of the login page, something that was difficult with the login.css overrides.

File: interface/globals.php [hunk 1]
```diff
@@ -361,11 +361,11 @@ function undoMagicQuotes($array, $topLevel=true) {
...
 }
 $login_filler_line = ' bgcolor="#f7f0d5" ';
-$logocode = "<img src='$web_root/sites/" . $_SESSION['site_id'] . "/images/login_logo.gif'>";
+$logocode = "<img class='img-responsive center-block' src='$web_root/sites/" . $_SESSION['site_id'] . "/images/login_logo.gif'>";
 // optimal size for the tiny logo is height 43 width 86 px
 // inside the open emr they will be auto reduced
-$tinylogocode1 = "<img class='tinylogopng' src='$web_root/sites/" . $_SESSION['site_id'] . "/images/logo_1.png'>";
-$tinylogocode2 = "<img class='tinylogopng' src='$web_root/sites/" . $_SESSION['site_id'] . "/images/logo_2.png'>";
+$tinylogocode1 = "<img class='img-responsive' src='$web_root/sites/" . $_SESSION['site_id'] . "/images/logo_1.png'>";
+$tinylogocode2 = "<img class='img-responsive' src='$web_root/sites/" . $_SESSION['site_id'] . "/images/logo_2.png'>";
```
Comment (bradymiller) [inline interface/globals.php]: Note the tinylogos were added by matrix. These are for 2 small logos which are also shown on the login screen and also within OpenEMR (in the Frames layout) on the top bar. So would ensure the size in both the login screen and the size within OpenEMR on the top bar do not get changed.

**Comment by bradymiller** [comment]
Is the logo file missing?

File: interface/login/login.php [hunk 2]
```diff
@@ -1,248 +1,247 @@
...
+*/
+
+$fake_register_globals = false;
+$sanitize_all_escapes = true;
+
+$ignoreAuth = true;
 include_once("../globals.php");
 include_once("$srcdir/sql.inc");
+
+$cookieValue = session_name() . " = " . session_id();
```
Comment (bradymiller) [inline interface/login/login.php]: Why is this done here? Wasn't it easier to follow when do this within the imsubmitted function?

File: interface/login/login.php [hunk 3]
```diff
@@ -1,248 +1,247 @@
...
 <html>
 <head>
 <?php html_header_show();?>
-<title><?php echo text($openemr_name) . " " . xlt('Login'); ?></title>
-<meta http-equiv="X-UA-Compatible" content="IE=edge" />
-<link rel=stylesheet href="<?php echo $css_header;?>" type="text/css">
-<link rel=stylesheet href="../themes/login.css" type="text/css">
-<link rel="shortcut icon" href="<?php echo $webroot; ?>/interface/pic/favicon.ico" />
-
-<script language='JavaScript' src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-min-1-4-3/index.js"></script>
```
Comment (bradymiller) [inline interface/login/login.php]: $groupTmp is not an array. This group code is a bit weird. Probably better off not touching it or else will need to test it etc.

File: interface/login/login.php [hunk 4]
```diff
@@ -1,248 +1,247 @@
...
+        <?php endif; ?>
+        return false;
+    }
+    </script>
 </head>
-<body onload="javascript:document.login_form.authUser.focus();" class="body_image">
-<span class="text"></span>
-<center>
-
-<form method="POST"
```
Comment (bradymiller) [inline interface/login/login.php]: mispelled $languags

**Comment by bradymiller** [comment]
Review complete. Screenshots look very nice. Can't test it since the logo file is missing.

File: interface/login/login.php (hunk 2 above)
Comment (robertdown) [inline interface/login/login.php]: When I was looking over that function I found it very difficult to work with it given the interpolation of PHP and JS so I moved the PHP out and only reference it via echo. It helps keep the code cleaner, in my opinion. Something I'd like to see more of

**Comment by robertdown** [comment]
Yep, it was late last night... just added the logo

File: interface/login/login.php (hunk 2 above)
Comment (bradymiller) [inline interface/login/login.php]: Just give it a quick comment then(ie. used in the imsubmitted js function).

**Comment by bradymiller** [comment]
Might as well get the favicon in there also (the blue swirl) to make this complete.

**Comment by robertdown** [comment]
Yep, was reading into that last night. There's a stupid number of icons that need to be made. If we wanted to be as fully compatible as possible for all browsers and OS's we'd need ~30 different icons!

For now I'm just gonna do the basic favicon.ico. Will get it in today or tomorrow.

**Comment by bradymiller** [comment]
Also, on the screenshots above, there appears to be no Language entry.

**Comment by robertdown** [comment]
Yeah, will add that... I had the languages set to just English in my local install

**Comment by bradymiller** [comment]
Lots of issues on my testing (I cleared the cache before using). Firefox 48.0 on Xubuntu 16.04.
See below screenshot.
(I also noted the language selector has a blank entry after the first entry which is very odd)
image

**Comment by robertdown** [comment]
Looks as if bootstrap isn't getting called for some reason

**Comment by bradymiller** [comment]
Also, the logo outlines are very rough appearing.

**Comment by robertdown** [comment]
Which logo? The new OpenEMR logo is a a vector image, it should be absolutely pristine under every possible resolution

**Comment by bradymiller** [comment]
There is no bootstrap include on the login page. Regarding the logo, am referring to the svg logo; perhaps it will get better with bootstrap included?

**Comment by robertdown** [comment]
How did I not include bootstrap??

As for the SVG, bootstrap shouldn't make a difference. I'll boot up a Linux VM and see if I can replicate

File: interface/login/login.php (hunk 3 above)
Comment (bradymiller) [inline interface/login/login.php]: This comment is not in the right place. will copy it again to the right place.

File: interface/login/login.php [hunk 5]
```diff
@@ -1,248 +1,247 @@
...
+        <form method="POST" id="loginForm" action="../main/main_screen.php?auth=login&site=<?php echo attr($_SESSION['site_id']); ?>" target="_top" name="login_form">
+            <div class="row">
+                <div class="col-sm-12">
+                    <div class="">
+                        <img class="img-responsive center-block" src="<?php echo $GLOBALS['webroot']?>/public/images/login-logo.svg" />
+                        <input type='hidden' name='new_login_session_management' value='1' />
+                        <?php
+                        // collect groups
+                        $res = sqlStatement("select distinct name from groups");
+                        $groupTmp = sqlFetchArray($res);
```
Comment (bradymiller) [inline interface/login/login.php]: $groupTmp is not an array. This group code is a bit weird. Probably better off not touching it or else will need to test it etc.

File: interface/login/login.php (hunk 4 above)
Comment (bradymiller) [inline interface/login/login.php]: this comment is not in the right place. will copy it to the right place.

File: interface/login/login.php [hunk 6]
```diff
@@ -1,248 +1,247 @@
...
+                                "ll.lang_description " .
+                                "FROM lang_languages AS ll " .
+                                "LEFT JOIN lang_constants AS lc ON lc.constant_name = ll.lang_description " .
+                                "LEFT JOIN lang_definitions AS ld ON ld.cons_id = lc.cons_id AND " .
+                                "ld.lang_id = ? " .
+                                "ORDER BY IF(LENGTH(ld.definition),ld.definition,ll.lang_description), ll.lang_id";
+                                $res3=SqlStatement($sql, array($mainLangID));
+                            }
+
+                            $languags = array();
```
Comment (bradymiller) [inline interface/login/login.php]: mispelled $languags

File: interface/login/login.php [hunk 7]
```diff
@@ -1,248 +1,247 @@
...
+                        foreach ($groupTmp as $group) {
+                            array_push($groups, $group);
+                        }
+                        if (count($groups) == 1) {
+                            echo "<input type='hidden' name='authProvider' value='" . attr($groups[0]) . "' />\n";
+                        }
+                        // collect default language id
+                        $res2 = sqlStatement("select * from lang_languages where lang_description = ?",array($GLOBALS['language_default']));
+                        for ($i = 0; $row = sqlFetchArray($res2); $i++) {
+                            $languages[] = $row;
```
Comment (bradymiller) [inline interface/login/login.php]: This is different than what you are using below for languages array, so would make this a different name. Maybe $default_language[]

**Comment by bradymiller** [comment]
Done with second review and also tested it. See comments above. Agree this will be nice looking when the bugs are ironed out.

**Comment by bradymiller** [comment]
Created an online demo for this here:
http://www.open-emr.org/wiki/index.php/Development_Demo#192.168.1.132

**Comment by robertdown** [comment]
Am back from my Red Cross work re: Hurricane Matthew and will put this on top of my priority list

**Comment by bradymiller** [comment]
image

**Comment by bradymiller** [comment]
Here's my screenshot. Note how:
1. rough edges of all the imaging in the logo
2. missing bootstrap

What is interesting is that if I zoom in, then the logo starts to look much nicer. Since this is a static image, wouldn't it be best to just use a png and then not need to worry about how svg gets rendered in a bunch of different web browsers and os's.

**Comment by bradymiller** [comment]
See here for the next revision of this PR:
https://github.com/openemr/openemr/pull/329

**Pull request closed without merging**